### PR TITLE
Add shadow to rounded view

### LINKED ISSUE #17: Add optional shadow to bottom view controller
The bottom view controller in the maps app has a drop shadow. We should see if we can provide this via the pullup controller similar to the dimmed view setup or if that should be the task of the bottom VC itself. In that latter case it might be nice to provide this option for the rounded view.

### PR DESCRIPTION
**Pull request by felixLam**
The shadow plays nicely with transparent views. 

Fixes #17 

simulator screen shot 09 07 2016 02 31 06

### CODE REVIEW

File: ISHPullUp/ISHPullUpRoundedView.m [hunk 1]
```diff
@@ -7,8 +7,73 @@
...
+    // The lip path is the basically our own bounds with top corners rounded using self.radius
+    // this defines what shape is casting a shadow
+    UIBezierPath *lipPath = [UIBezierPath bezierPathWithRoundedRect:self.bounds byRoundingCorners:UIRectCornerTopLeft | UIRectCornerTopRight cornerRadii:CGSizeMake(self.radius, self.radius)];
+    self.shadowPath = [lipPath CGPath];
+
+    /*
+     we want to mask the area that cast the shadow in order to not show a shadow there
+     we do not want to mask the area above (and outside of bounds)
+     we use the even-odd fill rule and add the entire bounds as rect plus the shadow length above
+     this yields the lip beeing covered twice (even) while everything above is covered once (odd)
```
Comment (mollidor) [inline ISHPullUp/ISHPullUpRoundedView.m]: being

File: ISHPullUp/ISHPullUpRoundedView.m [hunk 2]
```diff
@@ -7,8 +7,73 @@
...
+
+@implementation ISHPullUpRoundedTopShadowLayer
+
+- (instancetype)init {
+    self = [super init];
+    self.shapeMaskLayer = [CAShapeLayer new];
+    [self.shapeMaskLayer setFillColor:[[UIColor blackColor] CGColor]];
+    [self.shapeMaskLayer setFillRule:kCAFillRuleEvenOdd];
+    self.shadowOffset = CGSizeZero;
+    [self setMask:self.shapeMaskLayer];
```
Comment (mollidor) [inline ISHPullUp/ISHPullUpRoundedView.m]: seems `self.mask = ` would be more consistent with the setters above and below.

**Comment by mollidor** [comment]
👍

File: ISHPullUp/ISHPullUpRoundedView.m (hunk 2 above)
Comment (felixLam) [inline ISHPullUp/ISHPullUpRoundedView.m]: Converted everything in this m file to the `[ ..set:]`

**Pull request merged**